notebook/utils.py

```diff
@@ -140,7 +140,7 @@ def is_file_hidden_posix(abs_path, stat_res=None):
     if os.path.basename(abs_path).startswith('.'):
         return True
 
-    if stat_res is None:
+    if stat_res is None or stat.S_ISLNK(stat_res.st_mode):
```

Comment:
This should already be the `stat_res` for this path, at least according to the docstring. Do you mean to check ISLNK at another location, perhaps after calling `realpath`?

Reply:
In this case the docstring is wrong because _dir_model() passes the result of an lstat() call. To check if we are allowed to read the directory, we have to follow the symlink, though. That's why I called stat() again on the path.
There seems to be a good reason for using lstat() in _dir_model() (#2574).